### PR TITLE
Added documentation for System.Net.Security APIs targeted for 3.0

### PR DESCRIPTION
**Pull request by jozkee**
List of APIs documented:

* AuthenticatedStream
  * DisposeAsync
* NegotiateStream
  * DisposeAsync
* SslStream
  * DisposeAsync
  * ReadAsync(System.Memory{System.Byte},System.Threading.CancellationToken)
  * ReadAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)
  * ReadByte
  * WriteAsync(System.ReadOnlyMemory{System.Byte},System.Threading.CancellationToken)
  * WriteAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)

These changes contribute to System.Net.Security namespace, I am excluding changes related to 
`CipherSuitesPolicy`, `SslStream.NegotiatedCipherSuite` and `TlsCipherSuite` so I can tackle them in a separate PR.

### CODE REVIEW

**Comment by Jozkee** [comment]
@rpetrusha @mairaw @carlossanlop 
@dotnet/ncl

File: xml/System.Net.Security/AuthenticatedStream.xml [hunk 1]
```diff
@@ -182,9 +182,20 @@ The following example demonstrates displaying the properties of an authenticated
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Asynchronously releases the unmanaged and managed resources used by the <see cref="T:System.Net.Security.AuthenticatedStream" />.</summary>
+        <returns>A task that represents the asynchronous dispose operation.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ This method enables you to perform a resource-intensive dispose operation without blocking the main thread.
```
Comment (scalablecory) [inline xml/System.Net.Security/AuthenticatedStream.xml]: It's not too ambiguous here but "main" thread is a bit of a misnomer. Maybe just "current" thread?

Comment (Jozkee) [reply xml/System.Net.Security/AuthenticatedStream.xml]: Your comment makes sense to me; just consider that I took this paragraph from base class [`System.IO.Stream` documentation](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.disposeasync?view=netcore-3.0), perhaps we should modify that one as well.

Comment (carlossanlop) [reply xml/System.Net.Security/AuthenticatedStream.xml]: I wonder if we could reword the "enables you":

```suggestion
 This method allows performing resource-intensive dispose operations without blocking the main thread.
```

Regarding the main vs current, I think we should try to keep it in sync with IO.Stream, as much as possible.

File: xml/System.Net.Security/SslStream.xml [hunk 2]
```diff
@@ -2451,9 +2451,21 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Asynchronously releases the unmanaged and managed resources used by the <see cref="T:System.Net.Security.SsslStream" />.</summary>
```
Comment (rpetrusha) [inline xml/System.Net.Security/SslStream.xml]:
```suggestion
        <summary>Asynchronously releases the unmanaged and managed resources used by the <see cref="T:System.Net.Security.SslStream" />.</summary>
```

**Pull request merged**